From 446391b6254ee1eb7900b2e889f7a4dbd2323856 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 16 Jun 2020 17:04:52 +0200
Subject: GCC: do not use DCHECK_EQ/NE for checking PageNode::OpenedType

As OpenedType does not provide the insertion operator, required by
DCHECK_EQ and DCHECK_NE, we need to use the usual DCHECK instead.

Bug: 819294
Change-Id: I823e259b5703c700bbafc67e2db996ad0b07c2a0
---
 components/performance_manager/graph/page_node_impl.cc      | 6 +++---
 .../performance_manager/performance_manager_tab_helper.cc   | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/components/performance_manager/graph/page_node_impl.cc b/components/performance_manager/graph/page_node_impl.cc
index 14ec570a3fa2..8e0dc0617bd5 100644
--- a/components/performance_manager/graph/page_node_impl.cc
+++ b/components/performance_manager/graph/page_node_impl.cc
@@ -35,7 +35,7 @@ PageNodeImpl::PageNodeImpl(const WebContentsProxy& contents_proxy,
 PageNodeImpl::~PageNodeImpl() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   DCHECK_EQ(nullptr, opener_frame_node_);
-  DCHECK_EQ(OpenedType::kInvalid, opened_type_);
+  DCHECK(opened_type_ == OpenedType::kInvalid);
 }
 
 const WebContentsProxy& PageNodeImpl::contents_proxy() const {
@@ -251,7 +251,7 @@ void PageNodeImpl::SetOpenerFrameNodeAndOpenedType(FrameNodeImpl* opener,
   DCHECK(opener);
   DCHECK(graph()->NodeInGraph(opener));
   DCHECK_NE(this, opener->page_node());
-  DCHECK_NE(OpenedType::kInvalid, opened_type);
+  DCHECK(opened_type != OpenedType::kInvalid);
 
   auto* previous_opener = opener_frame_node_;
   auto previous_type = opened_type_;
@@ -269,7 +269,7 @@ void PageNodeImpl::SetOpenerFrameNodeAndOpenedType(FrameNodeImpl* opener,
 void PageNodeImpl::ClearOpenerFrameNodeAndOpenedType() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   DCHECK_NE(nullptr, opener_frame_node_);
-  DCHECK_NE(OpenedType::kInvalid, opened_type_);
+  DCHECK(opened_type_ != OpenedType::kInvalid);
 
   auto* previous_opener = opener_frame_node_;
   auto previous_type = opened_type_;
diff --git a/components/performance_manager/performance_manager_tab_helper.cc b/components/performance_manager/performance_manager_tab_helper.cc
index c221742b6086..768de49d9b44 100644
--- a/components/performance_manager/performance_manager_tab_helper.cc
+++ b/components/performance_manager/performance_manager_tab_helper.cc
@@ -396,7 +396,7 @@ void PerformanceManagerTabHelper::InnerWebContentsAttached(
     // using the window.open relationship if the embedded relationship is
     // severed.
   }
-  DCHECK_NE(PageNode::OpenedType::kInvalid, opened_type);
+  DCHECK(opened_type != PageNode::OpenedType::kInvalid);
   DCHECK(frame);
 
   PerformanceManagerImpl::CallOnGraphImpl(
-- 
2.20.1

